setup/lib/utils.js

```diff
@@ -0,0 +1,74 @@
+const { setFailed } = require('@actions/core');
+const os = require('os');
+const path = require('path');
+const execSync = require('child_process').execSync;
+const semver = require('semver');
+const core = require('@actions/core');
+
+// arch in [arm, x32, x64...] (https://nodejs.org/api/os.html#os_os_arch)
+// return value in [amd64, 386, arm]
+function mapArch(arch) {
+    const mappings = {
+        x32: '386',
+        x64: 'amd64'
+    };
+    return mappings[arch] || arch;
```

Comment:
We may need an exception for unknown `arch`

Reply:
Are you thinking of something like this?

```
function mapOS(os) {
    const mappings = {
        darwin: 'macOS',
        win32: 'windows'
    };

    if (os in mappings) {
        return mappings[os];
    } else {
        return 'Unknown OS';
    }
}
```

Reply:
I remove this func with a const with the value of `arm64`, but it might be better to replace it with a call to `os.Arch` to detect if it's amd or arm and append that to the `filename`.